### PR TITLE
SNOW-687973  Add view code links in API reference doc.

### PR DESCRIPTION
**Pull request by sfc-gh-stan**
Please answer these questions before submitting your pull requests. Thanks!

1. What GitHub issue is this PR addressing? Make sure that there is an accompanying issue to your PR.

   Fixes #[SNOW-687973](https://snowflakecomputing.atlassian.net/browse/SNOW-687973)

2. Fill out the following pre-review checklist:

   - [ ] I am adding a new automated test(s) to verify correctness of my new code
   - [ ] I am adding new logging messages
   - [ ] I am adding a new telemetry message
   - [ ] I am adding new credentials
   - [ ] I am adding a new dependency

3. Please describe how your code solves the related issue.

Enable linkcode support in our API reference doc. (The `[source]` link on the right)
Screen Shot 2022-11-02 at 11 30 21 AM

### CODE REVIEW

File: docs/source/conf.py [hunk 1]
```diff
@@ -89,3 +90,45 @@
...
+            source, lineno = inspect.getsourcelines(obj.fget)
+        else:
+            source, lineno = inspect.getsourcelines(obj)
+        linespec = f"#L{lineno}-L{lineno + len(source) - 1}"
+    except TypeError:
+            linespec = ""
+    return (
+        f"https://github.com/snowflakedb/snowpark-python/blob/"
+        f"main/{os.path.relpath(fn, start=os.pardir)}{linespec}"
+    )
```
Comment (sfc-gh-stan) [inline docs/source/conf.py:134]: Will fix

Comment (sfc-gh-yixie) [reply docs/source/conf.py:134]: How about using the release branch instead of the main?

Comment (sfc-gh-stan) [reply docs/source/conf.py:134]: `release-v1.0.0`?

Comment (sfc-gh-yixie) [reply docs/source/conf.py:134]: conf.py has variable `release`. You can use it

**Review by sfc-gh-jdu — APPROVED** [review]
This doc change looks awesome!

**Review by sfc-gh-mabrennan — APPROVED** [review]
Awesome! Thanks for adding this. Very cool.

Optionally, I have small a favour to ask. Because we'll be adding this improvement to the next update, could you please make a small change in the `/snowpark-python/main/docs/source/index.rst` file and remove the preview note at the top of the file? In other words, delete this:
```
.. sidebar:: Preview Feature --- Open

   Available to all accounts.
```

File: docs/source/conf.py [hunk 2]
```diff
@@ -89,3 +90,47 @@
...
+            source, lineno = inspect.getsourcelines(obj)
+        linespec = f"#L{lineno}-L{lineno + len(source) - 1}"
+    except TypeError:
+            linespec = ""
+    return (
+        f"https://github.com/snowflakedb/snowpark-python/blob/"
+        f"release-v{release}/{os.path.relpath(fn, start=os.pardir)}{linespec}"
+    )
+    
+
```
Comment (sfc-gh-sfan) [inline docs/source/conf.py:136]: I'm confused why github would complain no newline when it seems like we have two new lines 😂

**Review by sfc-gh-sfan — APPROVED** [review]

**Pull request merged**